src/components/common/form/FormDatePicker.tsx

```diff
@@ -0,0 +1,46 @@
+import { TextField } from '@mui/material'
+import { LocalizationProvider, DatePicker } from '@mui/x-date-pickers'
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
+import { format } from 'date-fns'
+import { useField, useFormikContext } from 'formik'
+import { i18n } from 'next-i18next'
+
+import { DATE_VALUE_FORMAT, getDateFormat } from 'common/util/date'
+
+/**
+ * MUI date picker to be connected with Formik. Propagates updates to the passed Formik field name
+ * @param name - name of the Formik field to bind
+ * @param label - prompt text
+ * @returns
+ */
+export default function FormDatePicker({ name, label }: { name: string; label: string }) {
+  const [field] = useField(name)
+  const { setFieldValue } = useFormikContext()
+
+  const dateViewFormat = getDateFormat(i18n?.language ?? '')
+  const mask = dateViewFormat.replace(new RegExp(/[^./]/g), '_')
+
+  const updateValue = (newValue: Date) => {
+    let formattedValue
+    try {
+      formattedValue = format(newValue, DATE_VALUE_FORMAT)
+    } catch {
+      formattedValue = field.value
+    } finally {
+      setFieldValue(name, formattedValue, true)
+    }
+  }
+
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <DatePicker
+        mask={mask}
+        inputFormat={dateViewFormat}
+        label={label}
+        value={field.value}
+        onChange={(newValue) => updateValue(newValue)}
+        renderInput={(params) => <TextField size="small" {...params} />}
```

Comment:
Maybe add a `prop` to change the size of the input rendered, so that it's not always `size=small`

Reply:
I'd leave such customization for whenever needed. I've only parametrized the needed parts as part of this task. What do you think?